db/sql/3_msar_custom_aggregates.sql

```diff
@@ -0,0 +1,127 @@
+/*
+This script defines all the necessary functions to be used for custom aggregates.
+
+Currently, we have the following custom aggregate(s):
+  - msar.peak_time(time): Calculate the 'average time' (interpreted as peak time) for a column.
+
+Refer to the official documentation of PostgreSQL custom aggregates to learn more.
+link: https://www.postgresql.org/docs/current/xaggr.html
+
+We'll use snake_case for legibility and to avoid collisions with internal PostgreSQL naming
+conventions.
+*/
+
+
+CREATE SCHEMA IF NOT EXISTS msar;
+
+CREATE OR REPLACE FUNCTION 
+msar.time_to_degrees(time_ TIME) RETURNS DOUBLE PRECISION AS $$/*
+Convert the given time to degrees (on a 24 hour clock, indexed from midnight).
+
+Examples:
+  00:00:00 =>   0
+  06:00:00 =>  90
+  12:00:00 => 180
+  18:00:00 => 270
+*/
+SELECT EXTRACT(EPOCH FROM time_) * 360 / 86400;
+$$ LANGUAGE SQL;
+
+
+CREATE OR REPLACE FUNCTION
+msar.degrees_to_time(degrees DOUBLE PRECISION) RETURNS TIME AS $$/*
+Convert given degrees to time (on a 24 hour clock, indexed from midnight).
+
+Steps:
+- First, the degrees is confined to range [0,360°)
+- Then the confined degrees is converted to time indexed from midnight.
+
+Examples:
+    0 => 00:00:00
+   90 => 06:00:00
+  180 => 12:00:00
+  270 => 18:00:00
+  540 => 12:00:00
+  -90 => 18:00:00
+
+Inverse of msar.time_to_degrees.
+*/
+SELECT MAKE_INTERVAL(secs => ((degrees::numeric % 360 + 360) % 360)::double precision * 86400 / 360)::time;
+$$ LANGUAGE SQL;
+
+
+CREATE OR REPLACE FUNCTION 
+msar.add_time_to_vector(sum_so_far point, time_ TIME) RETURNS point as $$/*
+Add the given time, converted to a vector on unit circle, to the vector given in first argument.
+
+We add a time to a point by
+- converting the time to a point on the unit circle.
+- adding that point to the point given in the first argument.
+
+Args:
+  sum_so_far: This is a point representing the sum of point so far.
+  time_: This is the time to be added to the running sum.
+
+Returns:
+  updated value of sum_so_far after adding the time_ to the previous value of sum_so_far.
+*/
+SELECT point(
+  sum_so_far[0] + sind(msar.time_to_degrees(time_)),
+  sum_so_far[1] + cosd(msar.time_to_degrees(time_))
+);
```

Comment:
```suggestion
WITH t(degrees) AS (SELECT msar.time_to_degrees(time_))
SELECT sum_so_far + point(sind(degrees), cosd(degrees)) FROM t;
```

I just noticed there's a cleaner way; it's also more performant (by a tiny bit; but it might matter since this gets run for every row)